pkg/log/zap/flags.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package zap contains helpers for setting up a new logr.Logger instance
+// using the Zap logging framework.
+
+package zap
+
+import (
+	"fmt"
+	"strings"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+type encoderConfigFunc func(*zapcore.EncoderConfig)
+
+type encoderValue struct {
+	set     bool
+	setFunc func(zapcore.Encoder)
+	str     string
+}
+
+func (v encoderValue) String() string {
+	return v.str
+}
+
+func (v encoderValue) Type() string {
+	return "encoder"
+}
+
+func (v *encoderValue) Set(e string) error {
+	v.set = true
+	val := strings.ToLower(e)
+	switch val {
+	case "json":
+		v.setFunc(newJSONEncoder())
+	case "console":
+		v.setFunc(newConsoleEncoder())
+	default:
+		return fmt.Errorf("invalid encoder value \"%s\"", e)
+	}
+	v.str = e
+	return nil
+}
+
+func newJSONEncoder(ecfs ...encoderConfigFunc) zapcore.Encoder {
+	encoderConfig := zap.NewProductionEncoderConfig()
+	for _, f := range ecfs {
+		f(&encoderConfig)
+	}
+	return zapcore.NewJSONEncoder(encoderConfig)
+}
+
+func newConsoleEncoder(ecfs ...encoderConfigFunc) zapcore.Encoder {
+	encoderConfig := zap.NewDevelopmentEncoderConfig()
+	for _, f := range ecfs {
+		f(&encoderConfig)
+	}
+	return zapcore.NewConsoleEncoder(encoderConfig)
+}
+
+type levelValue struct {
+	set     bool
+	setFunc func(zap.AtomicLevel)
+	str     string
+}
+
+func (v *levelValue) Set(l string) error {
+	v.set = true
+	lower := strings.ToLower(l)
+	switch lower {
+	case "debug":
+		v.setFunc(zap.NewAtomicLevelAt(zap.DebugLevel))
+	case "info":
+		v.setFunc(zap.NewAtomicLevelAt(zap.InfoLevel))
+	default:
+		return fmt.Errorf("invalid log level \"%s\"", l)
+	}
+	v.str = l
+	return nil
```

Comment:
In operator-sdk, we're allowing integer values as well (for increasing debug verbosity). We also hook this flag into the glog/klog verbosity as well.

https://github.com/operator-framework/operator-sdk/blob/633b92c888ad3eeb9704441b7f6d6bd15f933d0e/pkg/log/zap/flags.go#L105-L137

Do we want any of that here?